app.py

```diff
@@ -9,43 +11,80 @@
 parser.add_argument('name')
 parser.add_argument('function')
 
+params_schema = Schema({'name':str,'function':str})
+id_schema = Schema(All(str,Length(min=24,max=24,),Match(r'^[0123456789abcdef]*$')))
 
 class Thing(Resource):
-    def get(self, name):
-        thing = Model.get_by_name(name)
+    def get(self, _id):
+        try:
+            id_schema(_id)
+        except:
+            abort(400,message='invalid id')
+        try:
+            thing = Model.get_by_id(_id)
+        except:
+            abort(500, message='Unexpected Error '+str(sys.exc_info()[0]))
         if not thing:
-            abort(404, message='Thing {} not found'.format(name))
-        return vars(thing)
+            abort(404, message='Thing {} not found'.format(_id))
+        return jsonify(vars(thing))
 
-    def delete(self, name):
-        thing = Model.delete(name)
-        if not thing:
-            abort(404, message='Thing {} not found'.format(name))
-        return vars(thing), 204
+    def delete(self, _id):
+        try:
+            id_schema(_id)
+        except:
+            abort(400,message='invalid id')
+        try:
+            response = Model.delete(_id)
+        except:
+            abort(500, message='Unexpected Error '+str(sys.exc_info()[0]))
+        return jsonify(response)
 
-    def put(self, name):
+    def put(self, _id):
+        try:
+            id_schema(_id)
+        except:
+            abort(400,message='invalid id')
```

Comment:
same as above, do all of these within one try / except and handle the multiple exceptions that could be thrown in their own way